### PR TITLE
FIX(int-7914) Added error handling in case dataset/tables returns 404

### PR DESCRIPTION
**Pull request by Gonzalo-Avalos-Ribas**
Added error handling in case dataset/tables returns 404

### CODE REVIEW

File: src/steps/big-query/index.ts [hunk 1]
```diff
@@ -145,24 +148,28 @@ export async function fetchBigQueryModels(
...
-            await jobState.addEntity(modelEntity);
+        try {
+          await client.iterateBigQueryModels(
+            datasetEntity.name as string,
+            async (model) => {
+              const modelEntity = createBigQueryModelEntity(
+                model,
+                datasetEntity.id as string,
+              );
+              await jobState.addEntity(modelEntity);
```
Comment (jroblesx) [inline src/steps/big-query/index.ts:159]: Can we add a validation to ensure that the entity was created before adding the relationship?

File: src/steps/big-query/index.ts [hunk 2]
```diff
@@ -184,43 +191,67 @@ export async function fetchBigQueryTables(
...
-              createDirectRelationship({
-                _class: RelationshipClass.HAS,
-                from: datasetEntity,
-                to: tableEntity,
-              }),
-            );
-          },
-        );
+              await jobState.addEntity(tableEntity);
+              await jobState.addRelationship(
```
Comment (jroblesx) [inline src/steps/big-query/index.ts:220]: Can we add a validation to ensure that the entity was created before adding the relationship?

**Review by jroblesx — CHANGES_REQUESTED** [review]
I added some comments, I don't know if its absolutely necessary but I feel myself more comfortable with a validation before adding the relationship.

**Review by ndowmon — APPROVED** [review]

File: src/steps/big-query/index.ts (hunk 1 above)
Comment (ndowmon) [reply src/steps/big-query/index.ts:159]: I think this is out-of-scope for this change - this only has to do with error handling.

I also think that we can infer from the code that the entity was added before the relationship, so I don't think validation is necessary here.

**Review by ndowmon — APPROVED** [review]

**Review by jroblesx — APPROVED** [review]

**Pull request merged**